llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -434,6 +438,10 @@ PreservedAnalyses HWAddressSanitizerPass::run(Module &M,
     HWASan.sanitizeFunction(F, FAM);
 
   PreservedAnalyses PA = PreservedAnalyses::none();
+  // DominatorTreeAnalysis and LoopAnalysis are incrementally updated
+  // throughout this pass whenever SplitBlockAndInsertIfThen is called.
+  PA.preserve<DominatorTreeAnalysis>();
+  PA.preserve<LoopAnalysis>();
```

Comment:
Can we add PostDominatorTreeAnalysis here too? Seems like DomTreeUpdater also takes care of that.

Reply:
(And if not, please add a comment in the code on why not, because I think that's a natural question)

Reply:
Good point, added.